### PR TITLE
Kafka Plugin: Multi-cluster and multi-consumer per component

### PR DESCRIPTION
**Pull request by nirga**
## Hey, I just made a Pull Request!

Continued work on #3791. Added support for multiple clusters and multiple consumers per component. This included some breaking changes in the way the Kafka cluster is configured in `app-config.yaml`, but since the plugin was just added I'm not sure if it's a significant issue.

For representing components with multiple consumers I've chosen to split the data into multiple tables since I thought it might make more sense logically but I'm really not sure if it's the right approach now that I see the overall finished UI so I'm open for suggestions.

Screen Shot 2021-01-21 at 12 25 53

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

### CODE REVIEW

File: .changeset/fast-flowers-tickle.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-kafka': patch
+'@backstage/plugin-kafka-backend': patch
```
Comment (benjdlambert) [inline .changeset/fast-flowers-tickle.md:3]: Given that the config for this has changed I'm wondering if this is no longer a patch change?

Comment (benjdlambert) [reply .changeset/fast-flowers-tickle.md:3]: Also wondering if we should elaborate on the config changes and how to migrate in this changeset too. As it's a nice way to notify people of changes to different plugins and their migration path.

Comment (benjdlambert) [reply .changeset/fast-flowers-tickle.md:3]: Basically something that looks like this:

```diff
-  brokers:
-    - localhost:9092
+  clusters:
+    name: prod
+    brokers:
+      - localhost:9092

Comment (benjdlambert) [reply .changeset/fast-flowers-tickle.md:3]: Also this change would be good to document: https://github.com/backstage/backstage/pull/4182/files#diff-b6da313ecc5cb418e89d7f58e03271d96286b41ae2e8d34347eba39a8fad1adeL80-R82

Comment (nirga) [reply .changeset/fast-flowers-tickle.md:3]: Makes sense! updated.

Comment (benjdlambert) [reply .changeset/fast-flowers-tickle.md]: Minor is good! :) It's what it should be for breaking changes when the semver is <1.0.0

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**